honeybee_energy/material/gas.py

```diff
@@ -332,13 +332,15 @@ class EnergyWindowMaterialGas(_EnergyWindowMaterialGasBase):
         * specific_heat
         * density
         * prandtl
+        * user_data
     """
-    __slots__ = ('_gas_type',)
+    __slots__ = ('_gas_type','_user_data')
```

Comment:
Why are you adding `user_data` separately to each and every material class. Just implement it in `_EnergyMaterialBase` here:
https://github.com/ladybug-tools/honeybee-energy/blob/master/honeybee_energy/material/_base.py#L10
Then you don't need to specify it in all of these separate `__slots__` or use duplicated `@properties`. This also means that you won't need to set `self._user_data = None` in all of the `__init__` since it will all be handled in `_EnergyMaterialBase.__init__`.